code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm

```diff
@@ -524,92 +540,101 @@
 
 /datum/reagent/mental
-	name = "Experimental Antidepressant"
+	name = null //Just like alcohol
 	id = "mental"
 	description = "Some nameless, experimental antidepressant that you should obviously not have your hands on."
 	reagent_state = LIQUID
 	color = "#FFFFFF"
-	metabolism = 0.001
+	metabolism = 0.001 * REM
 	metabolism_min = 0
 	data = 0
+	scannable = 1
+	overdose = REAGENTS_OVERDOSE
 	taste_description = "bugs"
-	var/goodmessage = list("Your mind feels healthy.","You feel calm and relaxed.","The world seems like a better place now.") //Messages when all your brain traumas are cured.
-	var/badmessage = list("Your mind seems lost...","You feel agitated...","It feels like the world is out to get you...") //Messages when you still have at least one brain trauma it's suppose to cure.
-	var/worstmessage = list("Your mind starts to break down...","Things aren't what they seem...","You hate yourself...") //Messages when the user is at possible risk for more trauma
-	var/list/suppress_traumas  //List of brain traumas that the medication temporarily suppresses, with the key being the brain trauma and the value being the minimum dosage required to cure.
+	ingest_mul = 1
+	var/alchohol_affected = 1
+	var/min_dose = 1
+	var/messagedelay = ANTIDEPRESSANT_MESSAGE_DELAY
+	var/list/goodmessage = list() //Messages when all your brain traumas are cured.
+	var/list/badmessage = list() //Messages when you still have at least one brain trauma it's suppose to cure.
+	var/list/worstmessage = list() //Messages when the user is at possible risk for more trauma
+	var/list/suppress_traumas  //List of brain traumas that the medication temporarily suppresses, with the key being the brain trauma and the value being the minimum dosage required to cure. Negative values means that the trauma is temporarily gained on the value instead.
 	var/list/dosage_traumas //List of brain traumas that the medication permanently adds at these dosages, with the key being the brain trauma and the value being base percent chance to add.
 	var/list/withdrawal_traumas //List of withdrawl effects that the medication permanently adds during withdrawl, with the key being the brain trauma, and the value being the base percent chance to add.
-	var/messagedelay = ANTIDEPRESSANT_MESSAGE_DELAY
-	conflicting_reagents = list(/datum/reagent/alcohol/ = 5)
-	var/suppressing_reagents = list(/datum/reagent/synaptizine = 5) // List of reagents that suppress the withdrawal effects, with the key being the reagent and the vlue being the minimum dosage required to suppress.
+	var/list/suppressing_reagents = list() // List of reagents that suppress the withdrawal effects, with the key being the reagent and the vlue being the minimum dosage required to suppress.
 
 /datum/reagent/mental/affect_blood(var/mob/living/carbon/human/H, var/alien, var/removed)
 
-	if(!istype(H) || world.time < data)
+	if(!istype(H) || max_dose < min_dose || (world.time < data && volume > removed) || messagedelay == -1)
```

Comment:
`max_dose < min_dose` What is this? It seems like this is trying to enforce compile time mistakes at run time?  These are variables which shouldn't be changed at run time, so checking them isn't necessary. Remove.

Same here. What is this? `volume > removed`

Reply:
max_dose < min_dose and volume > removed prevents message startup/startdown spam if you're using something like a cigarette or similiar device that transfers a little bit of reagents at a time.
f in the case that a cigarette has a lower transfer rate than the metabolism rate, then message spam will occur since it's starting and stopping constantly.
This also prevents the whole code from working if the dosage is very small.